### DISCUSSION TITLE
Imports from Rez

### DISCUSSION THREAD

**Discussion by herronelou** [AcademySoftwareFoundation/rez#1621] [General]
During REZ TSC on 2024-01-18 it was asked to share parts of REZ that are being used through the API rather than the command line.

@JeanChristopheMorinPerso @maxnbk 

Here's my list:

```
from rez.config import config
from rez.exceptions import PackageRequestError
from rez.package_filter import PackageFilterList, RangeRule
from rez.package_repository import package_repository_manager
from rez.resolved_context import ResolvedContext
from rez.resolver import ResolverStatus
from rez.system import system
from rez.utils.formatting import PackageRequest, get_epoch_time_from_str
```

**Comment by JeanChristopheMorinPerso** [comment]
Thanks @herronelou! I'm kind of surprised to see `get_epoch_time_from_str`, but I think it makes sense since it's used by rez-env and rez-search directly. The rest (and even `get_epoch_time_from_str`) seems pretty standard.

That's great information for us. Thanks again!

**Reply by herronelou** [reply]
Yeah, we've extended the syntax to allow more ways to specify it, and are using it as a fallback. Now that I think about it, I don't even know if the fallback would work, I seem to recall that get_epoch_time_from_str only takes strings that start with a minus, and I don't know if our own argparse would not get confused that this is another arg (as it starts by -*), I seem to have hit that problem recently.
